### ISSUE TITLE
Allow user to disable gradient when doing remote execution

### ISSUE THREAD

**Issue by Butanium** [ndif-team/nnsight#251]
That would be nice to allow us to use bigger batch size when we don't need the gradient

**Comment by JadenFiotto-Kaufman** [comment]
@Butanium By default the params of the model do not require grad so assuming you dont set a tensor to require gradients its as if it was under `torch.no_grad()`

**Comment by clemneo** [comment]
It took quite a bit of troubleshooting and digging around on the Discord for me to realise that it is only remote execution that gradients aren't tracked — I had some code that was working locally, but not remotely. It seems that the only mention on the documentation is on the attribution patching tutorial, under ["Remote Attribution Patching"](https://nnsight.net/notebooks/tutorials/attribution_patching/#id1):

```
clean_out = []
corrupted_out = []
corrupted_grads = []

with llm.trace(remote = True) as tracer:
# Using nnsight's tracer.invoke context, we can batch the clean and the
# corrupted runs into the same tracing context, allowing us to access
# information generated within each of these runs within one forward pass

    with tracer.invoke(clean_tokens) as invoker_clean:
>>>>  # need to set requires grad to true for remote
>>>>    llm.model.layers[0].self_attn.o_proj.input.requires_grad = True
        # Gather each layer's attention
        for layer in llm.model.layers:
```

which can be quite easily missed in my opinion. I'm not sure where else would be the best place to mention this, but it feels like it should be mentioned somewhere else in the documentation so that it shows up when searching 'gradients'
